components/brave_shields/browser/ad_block_subscription_filters_provider.cc

```diff
@@ -27,27 +28,32 @@ AdBlockSubscriptionFiltersProvider::AdBlockSubscriptionFiltersProvider(
 AdBlockSubscriptionFiltersProvider::~AdBlockSubscriptionFiltersProvider() =
     default;
 
-void AdBlockSubscriptionFiltersProvider::LoadDATBuffer(
-    base::OnceCallback<void(bool deserialize, const DATFileDataBuffer& dat_buf)>
-        cb) {
+void AdBlockSubscriptionFiltersProvider::LoadFilterSet(
+    rust::Box<adblock::FilterSet>* filter_set,
+    base::OnceCallback<void()> cb) {
   base::ThreadPool::PostTaskAndReplyWithResult(
       FROM_HERE, {base::MayBlock()},
       base::BindOnce(&brave_component_updater::ReadDATFileData, list_file_),
       base::BindOnce(&AdBlockSubscriptionFiltersProvider::OnDATFileDataReady,
-                     weak_factory_.GetWeakPtr(), std::move(cb)));
+                     weak_factory_.GetWeakPtr(), std::move(cb), filter_set));
 }
 
 std::string AdBlockSubscriptionFiltersProvider::GetNameForDebugging() {
   return "AdBlockSubscriptionFiltersProvider";
 }
 
 void AdBlockSubscriptionFiltersProvider::OnDATFileDataReady(
-    base::OnceCallback<void(bool deserialize, const DATFileDataBuffer& dat_buf)>
-        cb,
+    base::OnceCallback<void()> cb,
+    rust::Box<adblock::FilterSet>* filter_set,
     const DATFileDataBuffer& dat_buf) {
-  auto metadata = adblock::read_list_metadata(dat_buf);
-  on_metadata_retrieved_.Run(metadata);
-  std::move(cb).Run(false, dat_buf);
+  auto result = (*filter_set)->add_filter_list(dat_buf);
+  if (result.result_kind == adblock::ResultKind::Success) {
+    on_metadata_retrieved_.Run(result.value);
+  } else {
+    LOG(ERROR) << "Subscription list parsing failed: "
```

Comment:
we should also ensure that the error message doesn't contain any potentially identifying info

Reply:
This only happens if a user subscribes to a list with malformed (i.e. non-UTF8) contents. We don't want to collect any crashdumps from here.

I added it so that there's _some_ visible reason for the failure, primarily to aid with local testing. The list already gets displayed with a generic "download failure" in `brave://settings/shields/adblock` if the update doesn't succeed.

Reply:
if this is primarily for local testing it should be a VLOG. `LOG(ERROR)` should generally be avoided